frontend/packages/console-shared/src/hooks/select-list.ts

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { OnSelect } from '@patternfly/react-table';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+
+export const useSelectList = <R extends K8sResourceCommon>(
+  data: R[],
+  visibleRows: Set<string>,
+  onRowSelected: (rows: R[]) => void,
+): {
+  onSelect: OnSelect;
+  selectedRows: Set<string>;
+  updateSelectedRows: (rows: R[]) => void;
+} => {
+  const [selectedRows, setSelectedRows] = React.useState<Set<string>>();
+
+  const onSelect = React.useCallback(
+    (_event, isSelected, rowIndex, rowData) => {
+      const uniqueUIDs: Set<string> = selectedRows ? new Set([...selectedRows]) : new Set<string>();
+      if (rowIndex === -1) {
+        if (isSelected) {
```

Comment:
Can be one liner.
We should also use `forEach`, not `map`,
And I dont think you need to call `has` before `delete`
```suggestion
        isSelected ? visibleRows.forEach((uid) => uniqueUIDs.add(uid)) : visibleRows.forEach((uid) => uniqueUIDs.delete(uid))
```